src/System.Reflection/tests/PropertyInfo/PropertyInfo_GetSetValueTests.cs

```diff
@@ -0,0 +1,374 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Xunit;
+using System;
+using System.Reflection;
+using System.Collections.Generic;
+
+#pragma warning disable 0414
+
+namespace System.Reflection.Tests
+{
+    public class PropertyInfoGetValueTests
+    {
+        public static void TestGetSet_PropertyA()
+        {
+            string propertyName = "PropertyA";
+            MyCoVariantTest obj = new MyCoVariantTest();
+            PropertyInfo pi = GetProperty(typeof(MyCoVariantTest), propertyName);
+
+            Assert.Equal(pi.Name, propertyName);
+
+            var value = pi.GetValue(obj);
+
+            Assert.NotNull(value);
+
+            string[] strs = new string[1];
+            strs[0] = "hello";
+
+            //set value
+            pi.SetValue(obj, strs);
+            value = pi.GetValue(obj);
+
+            string[] strs2 = (string[])value;
+
+            Assert.Equal(strs2[0], strs[0]);
+        }
+
+
+        [Fact]
+        public static void TestGetSet_PropertyB()
+        {
+            string propertyName = "PropertyB";
+            MyCoVariantTest obj = new MyCoVariantTest();
+            PropertyInfo pi = GetProperty(typeof(MyCoVariantTest), propertyName);
+
+            Assert.Equal(pi.Name, propertyName);
+
+            var value = pi.GetValue(obj);
+
+            Assert.Null(value);
+
+            string[] strs = new string[1];
+            strs[0] = "hello";
+
+            //set value
+            pi.SetValue(obj, strs);
+            value = pi.GetValue(obj);
+
+            string[] strs2 = (string[])value;
+
+            Assert.Equal(strs2[0], strs[0]);
+        }
+
+        [Fact]
+        public static void TestGetSet_Name()
+        {
+            string propertyName = "Name";
+            InterfacePropertyImpl obj = new InterfacePropertyImpl();
+            PropertyInfo pi = GetProperty(typeof(InterfacePropertyImpl), propertyName);
+
+            Assert.Equal(pi.Name, propertyName);
+
+            var value = pi.GetValue(obj, (Object[])null);
+
+            Assert.Null(value);
+
+            //set value
+            string strs1 = "hello";
+
+            pi.SetValue(obj, strs1);
+            value = pi.GetValue(obj);
+            string strs2 = (string)value;
+            Assert.Equal(strs2, strs1);
```

Comment:
I'm probably missing some subtle differences, but at first glance the above three tests all look identical (or close to identical), except with a different property name.  Do they all need to exist?  And if so, should this be a [Theory] that takes the property name as a string argument to the test?